File: 2533-bitwise-xor-of-all-pairings/2533-bitwise-xor-of-all-pairings.cs
public class Solution {
    public int XorAllNums(int[] nums1, int[] nums2) {
        if(nums1.Length % 2== 0 && nums2.Length % 2 == 0) return 0;
        int xorValue = 0;
        int ans = 0;
        for(int i = 0 ; i < nums2.Length; i++) xorValue ^= nums2[i];
        Console.WriteLine(xorValue);
        int n = nums1.Length;
        if(nums1.Length % 2 == 1 && nums2.Length % 2==0){
            return xorValue;
        }
        for(int i = 0 ;i < nums1.Length;i++){
            int x = xorValue ^ nums1[i];
            ans ^= x;
        }
        return ans;
    }
}

File: 2533-bitwise-xor-of-all-pairings/2533-bitwise-xor-of-all-pairings.cpp
class Solution {
public:
    int xorAllNums(vector<int>& nums1, vector<int>& nums2) {
        int n = nums1.size();
        int m = nums2.size();
        int xorValue1 = 0, xorValue2 = 0;
        for(auto u:nums1) xorValue1 ^= u;
        for(auto u:nums2) xorValue2 ^= u;
        if(n % 2 == 0 && m % 2 == 0) return 0;
        if(n % 2 == 1 && m % 2 == 0) return xorValue2;
        if(n % 2 == 0 && m % 2 == 1) return xorValue1;
        else  return (xorValue1 ^ xorValue2);
          
    }
};